src/NFTDriver.sol

```diff
@@ -55,7 +63,30 @@ contract NFTDriver is ERC721Burnable, ERC2771Context, Managed {
     /// @notice Get the ID of the next minted token.
     /// @return tokenId The token ID.
     function nextTokenId() public view returns (uint256 tokenId) {
-        return (uint256(driverId) << 224) + StorageSlot.getUint256Slot(_mintedTokensSlot).value;
+        return calcTokenIdWithSalt(address(0), _nftDriverStorage().mintedTokens);
+    }
+
+    /// @notice Calculate the ID of the token minted with salt.
+    /// @param minter The minter of the token.
+    /// @param salt The salt used for minting the token.
+    /// @return tokenId The token ID.
+    function calcTokenIdWithSalt(address minter, uint64 salt)
+        public
+        view
+        returns (uint256 tokenId)
+    {
+        tokenId = driverId;
+        tokenId = (tokenId << 160) | uint160(minter);
+        tokenId = (tokenId << 64) | salt;
+    }
+
+    /// @notice Checks if the salt has already been used for minting a token.
+    /// Each minter can use each salt only once, to mint a single token.
+    /// @param minter The minter of the token.
+    /// @param salt The salt used for minting the token.
+    /// @return isUsed True if the salt has been used, false otherwise.
+    function isSaltUsed(address minter, uint64 salt) public view returns (bool isUsed) {
```

Comment:
We can just make `exists` public

Reply:
We could also keep this function but build it on top of `exists`.

Reply:
I've looked into this and `exists` is reset after burning a token. If we want burning to be permanent, we can't use `exists`.

Reply:
Yes, you are right. I came to the conlusion if we `burn` it should be not possible to re-mint